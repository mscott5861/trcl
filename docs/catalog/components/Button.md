I'm just crazy about buttons
