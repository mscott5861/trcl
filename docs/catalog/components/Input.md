A basic Input component that maintains copies of a *real* and a *display* value (for cases in which masks are needed), and also maintains its own error state, which it reports upward to a parent **<Form/>** component.

```hint|directive
Inputs are made to be composable. Currently, additional functionality (such as validation, typeaheads, or masking) can be added by wrapping the component in the desired HOC.
```

# Props
```table
span: 6
rows:
  - Prop: activeLabel
    Required: no
    Type: string (hex/rgba/rgb)
    Description: The label displayed above the input field when active
  - Prop: activeLabelColor
    Required: no
    Type: string
    Description: The color of the activeLabel
  - Prop: bgColor
    Required: no
    Type: string (hex/rgba/rgb)
    Description: Background color of the input
  - Prop: borderless
    Required: no
    Type: boolean
    Description: If false, the input will have no border. If true, it will.
  - Prop: borderColor
    Required: no
    Type: string (hex/rgba/rgb)
    Description: Border color of the input in its default (non-errored) state
  - Prop: inputID
    Required: '**yes**'
    Type: string
    Description: A unique ID submitted by the form to identify the input's value
  - Prop: labelColor
    Required: no
    Type: string
    Description: The color of the inactive/blurred label of an empty input field
  - Prop: required
    Required: no
    Type: boolean
    Description: Whether the input contains a required field. Form will not submit with empty required fields.
```

# HOCs
