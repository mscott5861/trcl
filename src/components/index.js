import CodeBlock from './atomic/containers/CodeBlock'
import Copy from './atomic/text/Copy'
import H1 from './atomic/text/H1'
import H2 from './atomic/text/H2'
import H3 from './atomic/text/H3'
import HR from './atomic/embellishments/HR'
import HeroBanner from './atomic/images/HeroBanner'
import ImageWithCaption from './atomic/images/ImageWithCaption'
import Input from './atomic/text/Input'
import Leaflet from './atomic/containers/Leaflet'
import Link from './atomic/text/Link'
import MainContent from './atomic/containers/MainContent'
import Navbar from './atomic/containers/Navbar'
import PropsTable from './composite/containers/PropsTable'
import Sidebar from './atomic/containers/Sidebar'
import SingleColumnWithSidebar from './atomic/layouts/page-layouts/SingleColumnWithSidebar'
import SingleColumnWithSidebarAndNavbar from './atomic/layouts/page-layouts/SingleColumnWithSidebarAndNavbar'
import Table from './atomic/containers/Table'
import ThreeColumnLayout from './atomic/layouts/sublayouts/ThreeColumnLayout'
import TwoColumnLayout from './atomic/layouts/sublayouts/TwoColumnLayout'

export {
  CodeBlock,
  Copy,
  H1,
  H2,
  H3,
  HR,
  HeroBanner,
  ImageWithCaption,
  Input,
  Leaflet,
  Link,
  MainContent,
  Navbar,
  PropsTable,
  Sidebar,
  SingleColumnWithSidebar,
  SingleColumnWithSidebarAndNavbar,
  Table,
  ThreeColumnLayout,
  TwoColumnLayout,
}
