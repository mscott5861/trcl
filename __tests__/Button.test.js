import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { Button } from 'components'

Enzyme.configure({
  adapter: new Adapter()
});

describe('Button', () => {
  it('renders correctly', () => {
    const output = shallow(
      <Button 
        bgColor='#fff'
        label='Test Label'
        labelColor='#000'
        onClick={() => { console.log("Clicked"); }}
        />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

it('handles onClick event', () => {
  console.log = jest.fn();
  const output = shallow(
    <Button 
      onClick={() => { console.log("Clicked"); }}/>);
  output.simulate('click');
  expect(console.log).toHaveBeenCalledWith('Clicked');
});
