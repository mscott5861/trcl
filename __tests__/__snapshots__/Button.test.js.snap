// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<styled.button
  bgColor="#fff"
  onClick={[Function]}
  onFocus={[Function]}
  tabIndex="0"
  type="submit"
>
  <styled.p
    labelColor="#000"
  >
    Test Label
  </styled.p>
</styled.button>
`;
